### PR TITLE
Add Configuring GitHub OAuth section

### PR DESCRIPTION
**Pull request by vinokurig**
Signed-off-by: Igor Vinokur <ivinokur@redhat.com>

### What does this PR do?
Add Configuring GitHub OAuth section

### What issues does this PR fix or reference?
https://github.com/eclipse/che-theia/pull/497

### CODE REVIEW

File: src/main/pages/che-7/end-user-guide/proc_configuring-github-oauth.adoc [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// version-control
+
+[id="configuring-github-oauth_{context}"]
+== Configuring GitHub OAuth
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/proc_configuring-github-oauth.adoc]:
```suggestion
= Configuring GitHub OAuth
```

File: src/main/pages/che-7/end-user-guide/proc_configuring-github-oauth.adoc [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+OAuth for Github allows users to import projects using SSH addresses (`git@`), push to repositories, and use the *Pull Request* panel. To enable automatic SSH key upload to GitHub for users:
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/proc_configuring-github-oauth.adoc]:
```suggestion
OAuth for Github allows users to import projects using SSH addresses (`git@`), push to repositories, and use the *Pull Request* panel.
```

File: src/main/pages/che-7/end-user-guide/proc_configuring-github-oauth.adoc [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+. On *github.com*, click your user icon (top right).
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/proc_configuring-github-oauth.adoc:12]:
```suggestion
.Procedure

To enable automatic SSH key upload to GitHub for users:

. On *github.com*, click your user icon (top right).
```

File: src/main/pages/che-7/end-user-guide/proc_configuring-github-oauth.adoc [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+. Go to *Settings* > *Developer settings* > *OAuth Apps*.
+. Click the btn:[Register a new application] button.
+. In the *Application name* field, enter, for example, `Eclipse Che`.
+. In the *Homepage URL* field, enter `pass:q[http://__${CHE_HOST}__:__${CHE_PORT}__]`.
+. In the *Authorization callback URL* field, enter `pass:q[http://__${CHE_HOST}__:__${CHE_PORT}__/api/oauth/callback]`.
++
+image::git/github_oauth.png[]
+
+ifeval::["{project-context}" == "che"]
+. On OpenShift/Kubernetes, update the deployment configuration (see link:openshift-config.html[OpenShift configuration]).
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/proc_configuring-github-oauth.adoc]:
```suggestion
. On OpenShift or Kubernetes, update the deployment configuration (see link:https://www.eclipse.org/che/docs/che-6/openshift-config.html[OpenShift configuration]).
```

File: src/main/pages/che-7/end-user-guide/proc_configuring-github-oauth.adoc [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
++
+[subs=+quotes]
+----
+CHE_OAUTH_GITHUB_CLIENTID=__<your-github-client-id>__
+CHE_OAUTH_GITHUB_CLIENTSECRET=__<your-github-secret>__
+----
+
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/proc_configuring-github-oauth.adoc:30]:
```suggestion
endif::[]

```

File: src/main/pages/che-7/end-user-guide/proc_configuring-github-oauth.adoc [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+[NOTE]
+====
+* Substitute all occurrences of `_${CHE_HOST}_` and `_${CHE_PORT}_` with the URL and port of your Che installation.
+
+* Substitute `_<your-github-client-id>_` and `_<your-github-secret>_` with your GitHub client ID and secret.
+
+ifeval::["{project-context}" == "che"]
+* This configuration only applies to single-user deployments of Che.
+====
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/proc_configuring-github-oauth.adoc:40]:
```suggestion
endif::[]
====
```

**Review by rkratky — CHANGES_REQUESTED** [review]
Thanks, @vinokurig. I added some suggestions.

**Review by rkratky — APPROVED** [review]

**Comment by vinokurig** [comment]
@rkratky I've removed the `Pull request panel` it was added mistakenly

**Comment by rkratky** [comment]
> I've removed the Pull request panel it was added mistakenly

Oops, sorry, I misunderstood. Thanks.

Please, merge when ready.

**Pull request merged**